table/block_based_table_reader.cc

```diff
@@ -697,6 +732,7 @@ Status BlockBasedTable::Open(const ImmutableCFOptions& ioptions,
         if (rep->table_options.pin_l0_filter_and_index_blocks_in_cache &&
             level == 0) {
           rep->filter_entry = filter_entry;
+          rep->filter_entry.value->SetLevel(level);
```

Comment:
this causes a segfault when filter_policy is nullptr and both pin_l0_filter_and_index_blocks_in_cache/cache_index_and_filter_blocks are set.

repro:
`TEST_TMPDIR=/data/rate_limit_bench/  ./db_bench -benchmarks=fillrandom -pin_l0_filter_and_index_blocks_in_cache=true -cache_index_and_filter_blocks=true`